manifests/init.pp

```diff
@@ -49,6 +50,19 @@
     $dhcp_interfaces = $interfaces
   }
 
+  # See https://tools.ietf.org/html/draft-ietf-dhc-failover-12 for why BOOTP is
+  # not supported in the failover protocol. Relay agents *can* be made to work
+  # so $bootp can be explicitly set to true to override this default.
+  if $bootp == undef {
+    if $failover {
```

Comment:
I tend to avoid reliance on stdlib where possible and to make the code more obvious, but if consensus is to make it more terse, then I'll change it to pick().